protocol/contracts/libraries/Silo/LibLegacyTokenSilo.sol

```diff
@@ -0,0 +1,619 @@
+     * if they currently do not have any deposits to migrate, then this function 
+     * can be used to migrate their account to the new silo cheaply.
+     */
+   function _migrateNoDeposits(address account) internal {
+        AppStorage storage s = LibAppStorage.diamondStorage();
+        require(s.a[account].s.seeds == 0, "only for zero seeds");
+        
+        require(LibSilo.migrationNeeded(account), "no migration needed");
+
+        s.a[account].lastUpdate = s.season.stemStartSeason;
+    }
+
+
+    /** 
+     * @notice Migrates farmer's deposits from old (seasons based) to new silo (stems based).
+     * @param account Address of the account to migrate
+     * @param tokens Array of tokens to migrate
+     * @param seasons The seasons in which the deposits were made
+     * @param amounts The amounts of those deposits which are to be migrated
+     *
+     * @dev When migrating an account, you must submit all of the account's deposits,
+     * or the migration will not pass because the seed check will fail. The seed check
+     * adds up the BDV of all submitted deposits, and multiples by the corresponding
+     * seed amount for each token type, then compares that to the total seeds stored for that user.
+     * If everything matches, we know all deposits were submitted, and the migration is valid.
+     *
+     * Deposits are migrated to the stem storage system on a 1:1 basis. Accounts with
+     * lots of deposits may take a considerable amount of gas to migrate.
+     * 
+     * Returns seeds diff compared to stored amount, for verification in merkle check.
+     */
+    function _mowAndMigrate(
+        address account, 
+        address[] calldata tokens, 
+        uint32[][] calldata seasons,
+        uint256[][] calldata amounts
+    ) internal returns (uint256) {
+        //typically a migrationNeeded should be enough to allow the user to migrate, however
+        //for Unripe unit testing convenience, (Update Unripe Deposit -> "1 deposit, some", 
+        //"1 deposit after 1 season, all", and "2 deposit, all" tests), they cannot migrate
+        //since the test expects to add a Legacy deposit (which updates lastUpdated) and
+        //migrate in the same season, which doesn't work since lastUpdated is updated
+        //on deposit. By allow migration if balanceOfSeeds > 0, everything works smoothly.
+        //You would never be able to migrate twice since the old deposits would be removed already,
+        //and balanceOfSeeds would be 0 on 2nd migration attempt.
+        require((LibSilo.migrationNeeded(account) || balanceOfSeeds(account) > 0), "no migration needed");
+
+
+        //do a legacy mow using the old silo seasons deposits
+        updateLastUpdateToNow(account);
+        LibSilo.mintGrownStalk(account, balanceOfGrownStalkUpToStemsDeployment(account)); //should only mint stalk up to stemStartSeason
+        //at this point we've completed the guts of the old mow function, now we need to do the migration
```

Comment:
Worked on this here, lmk if this all look good:https://github.com/BeanstalkFarms/Beanstalk/commit/feba6f5740aeda82612900de4b19513f43ce3b2f

One thing I wasn't sure about is why I had to put `toUint128()` at the end of `perDepositData.grownStalk = perDepositData.grownStalk.add(stalkGrownSinceStemStartSeason).toUint128();` for example, since `grownStalk` is declared as uint128 in PerDepositData already. Without that it said `Type uint256 is not implicitly convertible to expected type uint128.`

Reply:
@pizzaman1337  This is because we used the typical safeMath for uint128, rather than LibSafeMath128 (`using SafeMath for uint128`). 

Fixed here: c4fd68cf8ef21601ac8b39565919b64e8d530c5a